examples/ftd_api_scripts/geodb_upload_update.py

```diff
@@ -0,0 +1,81 @@
+Code License, Version 1.1 (the "License"). A copy of the License
+can be found in the LICENSE.TXT file of this software or at
+https://developer.cisco.com/site/license/cisco-sample-code-license/
+All use of the material herein must be in accordance with the terms of
+the License. All rights not expressly granted by the License are
+reserved. Unless required by applicable law or agreed to separately in
+writing, software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied.
+"""
+
+import argparse
+import time
+from ftd_api_resources.access_token import get_access_token
+from ftd_api_resources.geodb_manual_update import send_geo_db_update_file, fetch_all_geo_db_jobs, fetch_geo_db_job_by_id
+
+
+def main():
+    # read the commandline arguments
+    parser = argparse.ArgumentParser(description='Upload a GeoDB update file then track the update job.')
+
+    parser.add_argument('-u', '--user-name', help='name of your API user', required=True, default='1')
+    parser.add_argument('-p', '--password', help='password of your API user', required=True, default='1')
+    parser.add_argument('-a', '--host', help='FTD host without port', required=True, default='1')
+    parser.add_argument('-n', '--port', help='FTD port number', required=True, default='1')
+    parser.add_argument('-f', '--file', help='path to update file', required=True, default='1')
+
+    args = parser.parse_args()
+
+    user_name = args.user_name
+    password = args.password
+    host = args.host
+    port = args.port
+    file = args.file
+
+    # send the update file
+    auth_token = get_access_token(host, port, user_name, password)
+    update_code = send_geo_db_update_file(host, port, auth_token, file)
+    if update_code != 200:
+        print("Upload failed. Aborting.")
+        exit(1)
+    print("Upload Successful.")
+
+    # get the uuid of the update job
+    jobs = fetch_all_geo_db_jobs(host, port, auth_token)
+    if jobs is None:
+        print('Failed to get update jobs. Aborting.')
+        exit(0)
+
+    current_job = jobs[0]
+    current_job_uuid = current_job.get('id')
+
+    # send periodic status checks for up to 5 minutes
+    try_count = 0
+    while try_count < 30:
+        current_job = fetch_geo_db_job_by_id(host, port, auth_token, current_job_uuid)
+        status = current_job.get('status')
+
+        # check the job status
+        if status == 'SUCCESS':
+            print('Update job was successful!')
+            exit(0)
+
+        if status == 'FAILED':
+            print('Update job was not successful.')
+            exit(0)
+
+        # job is still in progress; wait for 10 seconds and try again
+        print('Update job is still in progress. Check again in 10 seconds.')
+        time.sleep(10)
+        try_count += 1
+
+    print('Update job is still in progress after 5 minutes.')
+    exit(0)
+
+
+if __name__ == "__main__":
+    main()
```

Comment:
No change needed at this time. 
FYI, if api scripts get approved to use in automation, then incoming params will be parsed in this part of the script and passed to the function, which should be renamed from main() to geodb_upload_update() or similar, which should also return a success/fail value.  For a working example, see ftd_api_scripts/deploy.py. Or come up with your own approach if you can improve on it.